internal/service/searchdeployment/state_transition_search_deployment_test.go

```diff
@@ -73,15 +74,18 @@ func TestSearchDeploymentStateTransition(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			svc := mocksvc.NewDeploymentService(t)
-			ctx := context.Background()
+			m := mocksvc.NewAtlasSearchApi(t)
+			m.EXPECT().GetAtlasSearchDeployment(mock.Anything, mock.Anything, mock.Anything).Return(admin.GetAtlasSearchDeploymentApiRequest{ApiService: m})
+
 			for _, resp := range tc.mockResponses {
-				svc.On("GetAtlasSearchDeployment", ctx, dummyProjectID, clusterName).Return(resp.get()...).Once()
+				modelResp, httpResp, err := resp.get()
+				m.EXPECT().GetAtlasSearchDeploymentExecute(mock.Anything).Return(modelResp, httpResp, err).Once()
 			}
-			resp, err := searchdeployment.WaitSearchNodeStateTransition(ctx, dummyProjectID, "Cluster0", svc, testTimeoutConfig)
+			client := admin.APIClient{AtlasSearchApi: m}
+			resp, err := searchdeployment.WaitSearchNodeStateTransition(context.Background(), dummyProjectID, "Cluster0", &client, testTimeoutConfig)
 			assert.Equal(t, tc.expectedError, err != nil)
 			assert.Equal(t, responseWithState(tc.expectedState), resp)
-			svc.AssertExpectations(t)
+			m.AssertExpectations(t)
```

Comment:
this is not needed, it's done automatically for you if you create mocks with:  mocksvc.NewAtlasSearchApi